### DISCUSSION TITLE
calendar.get_events only give one event and not other events

### DISCUSSION THREAD

**Discussion by meet1919** [kuzmoyev/google-calendar-simple-api#139] [Q&A] [answered]
I have a service account through which I was success fully able to get the events from the GOOGLE CALENDAR API. But somehow I am only getting one event from the `.get_events()` method

```python
from gcsa.google_calendar import GoogleCalendar
from google.oauth2 import service_account

CALENDAR_ID=['xyz@gmail.com', 'abc@gmail.com']
SCOPES = ['https://www.googleapis.com/auth/calendar']
SERVICE_ACCOUNT_FILE = 'client_secrets2.json'

for CAL_ID in CALENDAR_ID:
    # authorize
    credentials = service_account.Credentials.from_service_account_file(SERVICE_ACCOUNT_FILE, scopes=SCOPES)
    calendar_auth = GoogleCalendar(CAL_ID, credentials=credentials)

    # get today and tomorrow's events
    calendar_events = calendar_auth.get_events(single_events=True, order_by='startTime')

    for event in calendar_events:
        print(event)
  
```

What should be causing this?

**Comment by kuzmoyev** [comment]
Could you please share how you got events from GOOGLE CALENDAR API, what events did you get, and what events you get with GCSA? `.get_events` should get you all events for the whole year starting now.

**Reply by meet1919** [reply]
Sorry, I wasn't clear, when I mentioned GOOGLE CALENDAR API, I meant GCSA. I have 3 google calendars. All of them worked properly a few days ago. But now from one calendar, I am getting 0 events, from another only one, and from 3 one it is very weird. When running the code in the python file (not inside the web app), I am getting all the events, when doing the same thing from the web app I am only getting the first event from all the events. This got me all confused. I can't really tell where the problem is. Is it in the API (I am getting all the events for only the 3rd calendar), or GCSA?

**Reply by kuzmoyev** [reply]
What do you mean by "web app"? Are python and gcsa versions the same everywhere?

**Reply by meet1919** [reply]
Yes, everything is same in webapp (in local) and also in the testing environment (a python file outside of the root directory of the webapp) as both share the same virtual environment

**Reply by kuzmoyev** [reply]
Are `client_secrets2.json` files the same?

**Reply by meet1919** [reply]
yes, this is same for both

**Reply by kuzmoyev** [reply]
What kind of web app is it? Please, check that the code is 100% identical. If not, please post the difference.
If it is the same, I really don't have any idea what might cause the discrepancy.

**Reply by meet1919** [reply]
It's a Django app, and yes code is identical. I am perplexed as you are.

**Reply by kuzmoyev** [reply]
Hi @meet1919, have you been able to find the source of the issue?

**Comment by meet1919** [comment, marked as answer]
The error was due to the character length set in database for storing the event summary `event.summary`

**Reply by kuzmoyev** [reply]
So it's unrelated to GCSA?
